test/test_utils.py

```diff
@@ -1198,6 +1219,68 @@ def test_build_eager(self):
+                            "Unequal inputs hashes should not match")
+
+    def test_verify_api_response_signature(self):
+        public_id = 'tests/logo.png'
+        test_version = 1
+        api_response_signature = '08d3107a5b2ad82e7d82c0b972218fbf20b5b1e0'
+
+        cloudinary.config(api_secret='X7qLTrsES31MzxxkxPPA-pAGGfU')
+
+        self.assertTrue(verify_api_response_signature(public_id, test_version, api_response_signature),
+                        "The response signature is valid for the same parameters")
+
+        self.assertFalse(verify_api_response_signature(public_id, test_version + 1, api_response_signature),
+                         "The response signature is invalid for the wrong version")
+
+        cloudinary.config(api_secret='b')
+
+    def test_verify_notification_signature(self):
+        valid_for = 60
+        signature = 'dfe82de1d9083fe0b7ea68070649f9a15b8874da'
+        body = '{"notification_type":"eager","eager":[{"transformation":"sp_full_hd/mp4","bytes":1055,' \
+               '"url":"http://res.cloudinary.com/demo/video/upload/sp_full_hd/v1533125278/dog.mp4",' \
+               '"secure_url":"https://res.cloudinary.com/demo/video/upload/sp_full_hd/v1533125278/dog.mp4"}],' \
+               '"public_id":"dog","batch_id":"9b11fa058c61fa577f4ec516bf6ee756ac2aefef095af99aef1302142cc1694a"}'
+
+        cloudinary.config(api_secret='X7qLTrsES31MzxxkxPPA-pAGGfU')
+        with patch('time.time', return_value=MOCKED_NOW):
+
+            valid_response_timestamp = time.time() - valid_for
+
+            test_message_part = "The notification signature is"
+
+            self.assertTrue(verify_notification_signature(body, valid_response_timestamp,
+                                                          signature,
+                                                          valid_for),
+                            "{} valid for matching and not expired signature".format(test_message_part))
+
+            self.assertFalse(verify_notification_signature(body, valid_response_timestamp,
+                                                           signature,
+                                                           valid_for - 1),
+                             "{} invalid for matching but expired signature".format(test_message_part))
+
+            self.assertFalse(verify_notification_signature(body, valid_response_timestamp,
+                                                           signature + 'chars'),
+                             "{} invalid for non matching and not expired signature".format(test_message_part))
+
+            self.assertFalse(verify_notification_signature(body, valid_response_timestamp,
+                                                           signature + 'chars',
+                                                           valid_for - 1),
+                             "{} invalid for non matching and expired signature".format(test_message_part))
+        cloudinary.config(api_secret='b')
```

Comment:
That is a bad practice, in case assertion fails, we never get to this line, use `with patch()`  for setting `api_key`